app/concerns/authentication/web.rb

```diff
@@ -6,54 +6,45 @@ module Authentication::Web
     protected
 
     def check_permanent_user
+      # transition users from old authentication to devise-based authentication
       return if logged_in?
       return unless cookies.signed[:user_id].present?
 
-      session[:user_id] = cookies.signed[:user_id]
-      set_user
+      # if the old cookie's corresponding user isn't found, log them out
+      unless (user = User.find_by(id: cookies.signed[:user_id]))
+        logout
+        return
+      end
```

Comment:
Separately, possibly:
```suggestion
      logout && return unless (user = User.find_by(id: cookies.signed[:user_id]))
```